### PR TITLE
vNext - DefaultSerializationOptions

### LINKED ISSUE #1246: vNext - CosmosClientOptions.SerializerOptions
Rename to `DefaultSerializerOptions` and return the default values.

### PR DESCRIPTION
**Pull request by ealsur**
# Pull Request Template

## Description

Following API review, this PR does two thing:

1. Rename `CosmosClientOptions.SerializerOptions` to `CosmosClientOptions.DefaultSerializerOptions` to better reflect the intent (customizing the default serializer)
2. Getting the `CosmosClientOptions.DefaultSerializerOptions` no longer returns `null` but the default instance with the default values of the `SerializerOptions`.

## Type of change

- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Closing issues

Closes #1246

### CODE REVIEW

File: Microsoft.Azure.Cosmos/azuredata/CosmosClientOptions.cs [hunk 1]
```diff
@@ -294,22 +295,22 @@ public IWebProxy WebProxy
...
         ///     }
         /// };
         /// 
         /// CosmosClient client = new CosmosClient("endpoint", "key", clientOptions);
         /// </example>
-        public CosmosSerializationOptions SerializerOptions
+        public CosmosSerializationOptions DefaultSerializerOptions
         {
-            get => this.serializerOptions;
+            get => this.serializerOptions ?? CosmosClientOptions.defaultSerializerOptions;
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/azuredata/CosmosClientOptions.cs]: What happens if users change the defaultSerializerOptions? The object it self it mutable.

Comment (ealsur) [reply Microsoft.Azure.Cosmos/azuredata/CosmosClientOptions.cs]: Good point

Comment (j82w) [reply Microsoft.Azure.Cosmos/azuredata/CosmosClientOptions.cs]: Is there any reason to make it immutable? Having mutable options always seems to cause issue.

Comment (ealsur) [reply Microsoft.Azure.Cosmos/azuredata/CosmosClientOptions.cs]: To make it immutable, that'd mean we need to offer the user a constructor to pass the parameters, or a Factory?

Comment (j82w) [reply Microsoft.Azure.Cosmos/azuredata/CosmosClientOptions.cs]: That's a lot or parameters to have in a constructor. The only other option I can think of is to add a flag to make it read only.

Comment (ealsur) [reply Microsoft.Azure.Cosmos/azuredata/CosmosClientOptions.cs]: That would mean we need to throw on set accessor if the flag is set? I changed the code now to return a new instance. That way the user can use it, even customize it if they want. I'm not sure throwing on set is friendly

Comment (j82w) [reply Microsoft.Azure.Cosmos/azuredata/CosmosClientOptions.cs]: What is the central .NET team guidance? The other solution I can think of is to break all the properties into smaller objects each having it's own constructor.

```
public sealed class CosmosClientOptions{

public SerializerOptions SerializerOptions {get;}
public ConnectionOptions ConnectionOptions {get;}
....
}
```

**Pull request merged**